File: CafeApi/Models/MenuItem.cs
using CafeApi.Enums;
using CafeApi.Interfaces;

namespace CafeApi.Models;

public class MenuItem : IEntity
{
    public int Id { get; private set; }
    public string Name { get; private set; } = null!;
    public ItemCategory Category { get; private set; }
    public decimal Price { get; private set; }
    public bool IsAvailable { get; private set; } = true;
    public string? Description { get; private set; }

    private MenuItem()
    {
    }

    public static MenuItem Create(string name, ItemCategory category, decimal price, string? description)
    {
        return new MenuItem
        {
            Name = name,
            Category = category,
            Price = price,
            Description = description,
            IsAvailable = true
        };
    }

    public void Update(
        string? name,
        decimal? price,
        bool? isAvailable,
        ItemCategory? category,
        string? description
    )
    {
        if (name is not null) Name = name;
        if (price is not null) Price = price.Value;
        if (isAvailable is not null) IsAvailable = isAvailable.Value;
        if (category is not null) Category = category.Value;
        if (description is not null) Description = description;
    }
}

File: CafeUI/src/types/menuItem.ts
import {ItemCategory} from "./enums/enums"

export interface MenuItemDto {
  id: number
  name: string
  category: keyof typeof ItemCategory
  price: number
  isAvailable: boolean
}

export interface CreateMenuItemDto {
  name: string
  category: keyof typeof ItemCategory
  price: number
  description?: string
}

export interface UpdateMenuItemDto {
  name?: string
  price?: number
  isAvailable?: boolean
}